packages/astro/src/actions/runtime/middleware.ts

```diff
@@ -0,0 +1,43 @@
+import { defineMiddleware } from '../../core/middleware/index.js';
+import { ApiContextStorage } from './store.js';
+import { formContentTypes, getAction } from './utils.js';
+import { callSafely } from './virtual/shared.js';
+import type { APIContext } from '../../@types/astro.js';
+
+export type Locals = {
+	_actionsInternal: {
+		getActionResult: APIContext['getActionResult'];
+	};
+};
+
+export const onRequest = defineMiddleware(async (context, next) => {
+	const locals = context.locals as Locals;
+	const { request } = context;
+	const contentType = request.headers.get('Content-Type');
+	locals._actionsInternal = {
+		getActionResult: () => undefined,
+	};
```

Comment:
nit: you might want to protect this object with a `Proxy` or `Object.create` from being rewritten from users.